### PR TITLE
Add step for no pauschbetrag

### PR DESCRIPTION
**Pull request by frederike-ramin**
# Short Description
- The step that is displayed if the person's merkzeichen and disability degree do not qualify them for either pauschbetrag or fahrkostenpauschale.

STILL TO DO: Add and adapt tests for the newly added preconditions of the pauschbetrag and no-pauschbetrag pages. I didn't get to that, I'm sorry.
And please have a look at the sonarcloud comments. Not sure why it's complaining. Sometimes we raise an error. 🤷‍♀️ 

# Changes
- Add the step and its preconditions
- Add preconditions for the PauschbetragStep
- Add integration tests
- Adapt the input parameters for the pauschbetrag and fahrkostenpauschale functions (has_pflegegrad should be string)
- Adapt method that collects data for the summary page to be able to show parts on the summary that do not have a given value in the forms

# Feedback
- Do you think the commented-out tests are a good idea? I think they'd help me a lot if I'd implement the next ticket. wdyt?

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: webapp/client/cypress/integration/noPauschbetragPersonA.spec.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            person_a_has_pflegegrad: "no",
+          });
+          cy.visit("/lotse/step/person_a_no_pauschbetrag");
+        });
+
+        it("Should link forward to person b page", () => {
+          cy.get("a").contains("Weiter").click();
+          cy.url().should("include", "/lotse/step/person_b");
+        });
+      });
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/noPauschbetragPersonA.spec.js:123]: Personally, I think this context should be outside the other context because it does not keep any of the data. However, if you think that this makes more sense because it structures the tests better, that would be fine with me

File: webapp/client/cypress/integration/personAHasDisability.spec.js [hunk 2]
```diff
@@ -31,9 +31,10 @@ describe("PersonAHasDisability", () => {
       cy.get("#person_a_has_disabilityno").should("not.be.checked");
     });
 
-    it("Should link forward to telephone number page", () => {
+    it("Should link forward to person a merkzeichen page", () => {
       cy.get("button[type=submit]").click();
       cy.url().should("include", "/lotse/step/telephone_number");
+      // TODO cy.url().should("include", "/lotse/step/merkzeichen_person_a");
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/personAHasDisability.spec.js:37]: This is already part of main and we should delete this line when merging

File: webapp/client/cypress/integration/personBHasDisability.spec.js [hunk 3]
```diff
@@ -39,9 +39,10 @@ describe("PersonBHasDisability", () => {
       cy.get("#person_b_has_disabilityno").should("not.be.checked");
     });
 
-    it("Should link forward to telephone number page", () => {
+    it("Should link forward to person b merkzeichen page", () => {
       cy.get("button[type=submit]").click();
       cy.url().should("include", "/lotse/step/telephone_number");
+      // TODO cy.url().should("include", "/lotse/step/merkzeichen_person_b");
```
Comment (JulianRoesner) [inline webapp/client/cypress/integration/personBHasDisability.spec.js:45]: Same as above

File: webapp/client/src/components/FormFieldRadioGroup.js [hunk 4]
```diff
@@ -19,6 +19,7 @@ const Radio = styled.div`
   }
 
   input[type="radio"] + label {
+    white-space: pre-wrap;
```
Comment (JulianRoesner) [inline webapp/client/src/components/FormFieldRadioGroup.js:22]: Why is this added?

File: webapp/client/src/pages/PauschbetragPage.js [hunk 5]
```diff
@@ -102,7 +99,9 @@ export default function PauschbetragPage({
 }
 
 PauschbetragPage.propTypes = {
-  stepHeader: stepHeaderPropType.isRequired,
+  stepHeader: PropTypes.shape({
+    title: PropTypes.string,
+  }).isRequired,
```
Comment (JulianRoesner) [inline webapp/client/src/pages/PauschbetragPage.js:104]: Why don't we keep the stepHeaderProptye?

File: acceptance_tests/cypress/integration/lotse_flow.spec.js [hunk 6]
```diff
@@ -249,6 +249,17 @@ context('Acceptance tests', () => {
...
+        it('Enter disability', () => {
+            cy.visit('/lotse/step/person_a_has_disability?link_overview=True')
+
+            // Set disability status to yes
+            cy.get('#person_a_has_disabilityyes').check().should('be.checked')
+            cy.get(submitBtnSelector).click()
+
+            // Select pauschbetrag
+            cy.get('#person_a_requests_pauschbetragyes').check().should('be.checked')
+        })
```
Comment (JulianRoesner) [inline acceptance_tests/cypress/integration/lotse_flow.spec.js:261]: I am not sure if this should be it's own path, wdyt?

File: webapp/app/forms/flows/lotse_flow.py [hunk 7]
```diff
@@ -350,15 +354,17 @@ def _collect_data_for_step(self, step, form_data, missing_fields=None):
...
+            if missing_fields and attr in missing_fields:
                 field = getattr(step.form, attr)
                 label = field.kwargs['render_kw']['data_label']
                 step_data[label] = _l('form.lotse.missing_mandatory_field')
+            elif hasattr(step.form, attr):
+                field = getattr(step.form, attr)
+                # TODO: When the summary page is refactored we should merge _generate_value_representation & get_overview_value_representation
+                label, value = self._generate_value_representation(field, form_data.get(attr))
+                value = step.get_overview_value_representation(value)
+                if value:
```
Comment (JulianRoesner) [inline webapp/app/forms/flows/lotse_flow.py:366]: Why did the order change?

File: webapp/app/forms/steps/lotse/fahrkostenpauschbetrag.py [hunk 8]
```diff
@@ -1,9 +1,11 @@
...
+    if has_pflegegrad == 'yes' or has_merkzeichen_bl or has_merkzeichen_tbl or has_merkzeichen_tbl \
+            or has_merkzeichen_h or has_merkzeichen_ag:
         return 4500
-    elif disability_degree >= 80 or (has_merkzeichen_g and disability_degree >= 70):
+    elif disability_degree and (disability_degree >= 80 or (has_merkzeichen_g and disability_degree >= 70)):
         return 900
-
     else:
-        return None
+        return 0
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/fahrkostenpauschbetrag.py:11]: Why not do it like this?

```python
def calculate_fahrkostenpauschbetrag(has_pflegegrad: str = None, disability_degree: int = 0,
                                       has_merkzeichen_bl: bool = False, has_merkzeichen_tbl: bool = False,
                                       has_merkzeichen_h: bool = False, has_merkzeichen_ag: bool = False,
                                       has_merkzeichen_g: bool = False):
if has_pflegegrad == 'yes' or has_merkzeichen_bl or has_merkzeichen_tbl or has_merkzeichen_tbl \
              or has_merkzeichen_h or has_merkzeichen_ag:
          return 4500
elif disability_degree >= 80 or (has_merkzeichen_g and disability_degree >= 70):
          return 900
```

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 9]
```diff
@@ -1,16 +1,41 @@
...
+from app.forms.steps.step import SectionLink
+from app.forms.steps.lotse.lotse_step import LotseFormSteuerlotseStep
+from app.forms.steps.lotse.personal_data import PersonAHasDisabilityPrecondition, PersonBHasDisabilityPrecondition, ShowPersonBPrecondition, StepFamilienstand, get_number_of_users
+
+
+from app.forms.steps.lotse.personal_data import get_number_of_users
+from app.model.components import NoPauschbetragProps
+
+
+def calculate_pauschbetrag(has_pflegegrad: str = None, disability_degree: int = None, has_merkzeichen_bl: bool = False,
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:23]: If we set the disability degree to zero here, we do not need to do the check below

File: webapp/app/forms/steps/lotse/fahrkostenpauschbetrag.py [hunk 10]
```diff
@@ -1,9 +1,11 @@
-def calculate_fahrkostenpauschbetrag(has_pflegegrad=False, disability_degree=None, has_merkzeichen_bl=False, has_merkzeichen_tbl=False,
-                                     has_merkzeichen_h=False, has_merkzeichen_ag=False, has_merkzeichen_g=False):
-    if has_pflegegrad or has_merkzeichen_bl or has_merkzeichen_tbl or has_merkzeichen_tbl or has_merkzeichen_h or has_merkzeichen_ag:
+def calculate_fahrkostenpauschbetrag(has_pflegegrad: str = None, disability_degree: int = None,
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/fahrkostenpauschbetrag.py:1]: Kind of funny that we have to change that function in the end as we discussed twice if we could trust the input data

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 11]
```diff
@@ -28,4 +53,328 @@ def calculate_pauschbetrag(has_pflegegrad=False, disability_degree=None, has_mer
...
+            if values['person_b_disability_degree'] \
+            else None
+        return values
+
+
+class PersonAHasPflegegradSetPrecondition(DisabilityModel):
+    _step_to_redirect_to = StepFamilienstand.name  # TODO: StepPersonAMerkzeichen.name
+    _message_to_flash = _l('form.lotse.skip_reason.has_not_filled_pflegegrad')
+
+    person_a_has_pflegegrad: str
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:95]: Why do we have to add this Precondition? Couldn't this be the hasMerkzeichenData Precondition?

Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/pauschbetrag.py:95]: Or should we use this for the MerkzeichenPrecondition because the `has_pflegegrad` is always required

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 12]
```diff
@@ -28,4 +53,328 @@ def calculate_pauschbetrag(has_pflegegrad=False, disability_degree=None, has_mer
...
+                'show_overview_button': bool(self.render_info.overview_url),
+            },
+            pauschbetrag=self.get_pauschbetrag(),
+            fields=form_fields_dict(self.render_info.form),
+            prev_url=self.render_info.prev_url
+        ).camelized_dict()
+
+        # Humps fails to camelize individual letters correctly, so we have to fix it manually.
+        # (A fix exists but hasn't been released at the time of writing: https://github.com/nficano/humps/issues/61)
+        props_dict['fields']['personARequestsPauschbetrag'] = props_dict['fields'].pop('personA_requestsPauschbetrag')
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:244]: We need to remove this once #532 has been merged

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 13]
```diff
@@ -28,4 +53,328 @@ def calculate_pauschbetrag(has_pflegegrad=False, disability_degree=None, has_mer
...
+            has_merkzeichen_tbl=values.get('person_b_has_merkzeichen_tbl', False),
+            has_merkzeichen_h=values.get('person_b_has_merkzeichen_h', False)
+        )
+
+        if pauschbetrag_claim != 0 or fahrkostenpauschbetrag_claim != 0:
+            raise ValidationError
+        return values
+
+
+class StepPauschbetragPersonA(LotseFormSteuerlotseStep):
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:197]: Why does this also add StepPauschbetrag?

File: webapp/app/forms/steps/lotse/personal_data.py [hunk 14]
```diff
@@ -523,4 +523,29 @@ def render(self):
...
+        if value != 'yes':
+            raise ValidationError
+        return value
+
+
+class PersonBHasDisabilityPrecondition(BaseModel):
+    _step_to_redirect_to = StepPersonBHasDisability.name
+    _message_to_flash = _l('form.lotse.skip_reason.has_no_disability')
+    
+    person_b_has_disability: str
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/personal_data.py:545]: Probably exchange those with the ones on main

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+
+    def test_if_required_value_is_given_then_validation_should_be_success(self, new_test_request_context,
+                                                                          valid_stored_data):
+        data = MultiDict({'person_b_requests_pauschbetrag': 'no'})
+
+        with new_test_request_context(method='POST', form_data=data, stored_data=valid_stored_data):
+            step = LotseStepChooser().get_correct_step(StepPauschbetragPersonB.name, True, ImmutableMultiDict(data))
+
+            form = step.render_info.form
+            assert form.validate() is True
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:64]: Could we also test for a failed validation?

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+
+    def test_if_required_precondition_show_person_b_is_not_satisfied_return_should_be_a_redirect_to_familienstand(self, new_test_request_context):
+        data = MultiDict({
+            'person_b_has_disability': 'yes',
+            'person_b_requests_pauschbetrag': 'yes'
+        })
+        with new_test_request_context(form_data=data):
+            step = LotseStepChooser().get_correct_step(
+                StepPauschbetragPersonB.name, True, ImmutableMultiDict(data))
+            assert step.redirection_step_name == 'familienstand'
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:90]: We test those in the functional tests in #547. I think we could remove those here

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+            step = LotseStepChooser().get_correct_step(
+                StepPauschbetragPersonA.name, True, ImmutableMultiDict(data))
+            assert step.redirection_step_name == 'person_a_has_disability'
+            
+    def test_if_precondition_person_a_has_disability_yes_is_not_satisfied_return_should_be_a_redirect_to_person_a_has_disability(self, new_test_request_context):
+        data = MultiDict({'person_a_has_disability':'no'})
+        with new_test_request_context(form_data=data):
+            step = LotseStepChooser().get_correct_step(
+                StepPauschbetragPersonA.name, True, ImmutableMultiDict(data))
+            assert step.redirection_step_name == 'person_a_has_disability'
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:41]: Same as below

**Review by JulianRoesner — COMMENTED** [review]
> Do you think the commented-out tests are a good idea? I think they'd help me a lot if I'd implement the next ticket. wdyt?

I do not mind. However, I think we could merge #547 before and then we would not need those todo comments

In general, I think there are still some open points to discuss

File: webapp/app/model/form_data.py [hunk 18]
```diff
@@ -155,6 +157,28 @@ def account_holder_must_be_declared(cls, v, values):
...
+    @validator('person_a_requests_pauschbetrag', always=True)
+    def required_if_person_a_has_pauschbetrag_claim(cls, v, values):
+        if not v:
+            try:
+                from app.forms.steps.lotse.pauschbetrag import PersonAHasPauschbetragClaimPrecondition
+                PersonAHasPauschbetragClaimPrecondition.parse_obj(values)
+                raise MissingError  # has pauschbetrag claim
+            except ValidationError:
+                return v  # has no pauschbetrag claim
+        return v
```
[the remark this answers is not in the document]
Comment (SannyNguyenHung) [reply webapp/app/model/form_data.py:169]: Done!

File: webapp/app/model/form_data.py [hunk 19]
```diff
@@ -155,6 +157,28 @@ def account_holder_must_be_declared(cls, v, values):
...
+    @validator('person_b_requests_pauschbetrag', always=True)
+    def required_if_person_b_has_pauschbetrag_claim(cls, v, values):
+        if not v:
+            try:
+                from app.forms.steps.lotse.pauschbetrag import PersonBHasPauschbetragClaimPrecondition
+                PersonBHasPauschbetragClaimPrecondition.parse_obj(values)
+                raise MissingError  # has pauschbetrag claim
+            except ValidationError:
+                return v  # has no pauschbetrag claim
+        return v
```
[the remark this answers is not in the document]
Comment (SannyNguyenHung) [reply webapp/app/model/form_data.py:180]: Done

File: webapp/app/forms/steps/lotse/fahrkostenpauschbetrag.py (hunk 8 above)
Comment (JulianRoesner) [reply webapp/app/forms/steps/lotse/fahrkostenpauschbetrag.py]: Because if the disability_degree is not set this would not work

File: webapp/tests/forms/steps/lotse/test_has_disability.py [hunk 20]
```diff
@@ -96,4 +96,57 @@ def test_if_person_b_has_disability_set_yes_then_do_not_raise_validation_error(s
...
+class TestStepDisabilityPersonAValidation:
+    def test_if_required_value_is_given_then_validation_should_be_success(self, new_test_request_context):
+        data = MultiDict({'person_a_has_disability': 'yes'})
+        with new_test_request_context(form_data=data):
+            step = LotseStepChooser().get_correct_step(
+                StepDisabilityPersonA.name, True, ImmutableMultiDict(data))
+            form = step.render_info.form
+            assert form.validate() is True
+
+    def test_if_required_value_is_not_give_validation_should_be_failure(self, new_test_request_context):
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_has_disability.py:111]: Maybe we could rename it to:
```suggestion
    def test_if_required_value_is_not_given_then_validation_should_be_failure(self, new_test_request_context):
```

File: webapp/tests/forms/steps/lotse/test_has_disability.py [hunk 21]
```diff
@@ -96,4 +96,57 @@ def test_if_person_b_has_disability_set_yes_then_do_not_raise_validation_error(s
...
+        data = MultiDict()
+        with new_test_request_context(form_data=data):
+            step = LotseStepChooser().get_correct_step(
+                StepDisabilityPersonA.name, True, ImmutableMultiDict(data))
+            form = step.render_info.form
+            assert form.validate() is False
+
+
+class TestStepDisabilityPersonBValidation:
+    def test_if_person_b_has_disability_is_given_validation_should_be_true(self, new_test_request_context):
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_has_disability.py:121]: Could we rename it:
```suggestion
    def test_if_person_b_has_disability_is_given_then_validation_should_be_true(self, new_test_request_context):
```

File: webapp/tests/forms/steps/lotse/test_pauschbetrag.py [hunk 22]
```diff
@@ -184,6 +190,73 @@ def test_if_merkzeichen_h_and_no_pflegegrad_set_then_return_7400_for_all_disabil
             calculated_pauschbetrag = calculate_pauschbetrag(**params, disability_degree=disability_degree)
             assert calculated_pauschbetrag == 7400
 
+    def test_if_no_parameters_then_return_zero(self):
+        calculated_pauschbetrag = calculate_pauschbetrag()
+        assert calculated_pauschbetrag == 0
+        
+    def test_if_disability_degree_under_20_then_return_zero(self):
+        calculated_pauschbetrag = calculate_pauschbetrag(disability_degree=19)
+        assert calculated_pauschbetrag == 0
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag.py:199]: I am wondering if we might need to test this also with the other params (has_pflegegrad etc.).

File: webapp/tests/forms/steps/lotse/test_pauschbetrag.py [hunk 23]
```diff
@@ -184,6 +190,73 @@ def test_if_merkzeichen_h_and_no_pflegegrad_set_then_return_7400_for_all_disabil
...
+    def test_if_calculate_pauschbetrag_returns_number_other_than_zero_then_raise_validation_error(self):
+        with patch('app.forms.steps.lotse.pauschbetrag.calculate_pauschbetrag', MagicMock(return_value=1)):
+            with pytest.raises(ValidationError):
+                HasNoPauschbetragOrFahrkostenpauschbetragClaimPersonAPrecondition.parse_obj({})
+
+    def test_if_calculate_fahrkostenpauschbetrag_returns_number_other_than_zero_then_raise_validation_error(self):
+        with patch('app.forms.steps.lotse.pauschbetrag.calculate_fahrkostenpauschbetrag',
+                   MagicMock(return_value=1)):
+            with pytest.raises(ValidationError):
+                HasNoPauschbetragOrFahrkostenpauschbetragClaimPersonAPrecondition.parse_obj({})
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag.py:240]: Personally, I would have expected another test: `test_if_calculate_pauschbetrag_and_fahrkostenpauschbetrag_return number_other_than_zero_then_raise_validation_error`. WDYT?

File: webapp/tests/forms/steps/lotse/test_pauschbetrag.py [hunk 24]
```diff
@@ -184,6 +190,73 @@ def test_if_merkzeichen_h_and_no_pflegegrad_set_then_return_7400_for_all_disabil
...
+    def test_if_calculate_pauschbetrag_returns_number_other_than_zero_then_raise_validation_error(self):
+        with patch('app.forms.steps.lotse.pauschbetrag.calculate_pauschbetrag', MagicMock(return_value=1)):
+            with pytest.raises(ValidationError):
+                HasNoPauschbetragOrFahrkostenpauschbetragClaimPersonBPrecondition.parse_obj({})
+
+    def test_if_calculate_fahrkostenpauschbetrag_returns_number_other_than_zero_then_raise_validation_error(self):
+        with patch('app.forms.steps.lotse.pauschbetrag.calculate_fahrkostenpauschbetrag',
+                   MagicMock(return_value=1)):
+            with pytest.raises(ValidationError):
+                HasNoPauschbetragOrFahrkostenpauschbetragClaimPersonBPrecondition.parse_obj({})
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag.py:259]: Same as above

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py [hunk 25]
```diff
@@ -16,31 +20,28 @@ def test_if_person_a_requests_pauschbetrag_is_given_then_validation_should_be_su
...
+    def test_if_precondition_person_a_has_disability_yes_is_not_satisfied_return_should_be_a_redirect_to_person_a_has_disability(self, new_test_request_context):
+        data = MultiDict({'person_a_has_disability':'no'})
+        with new_test_request_context(form_data=data):
             step = LotseStepChooser().get_correct_step(
-                StepPauschbetragPersonA.name, True, ImmutableMultiDict(form_data))
-            form = step.render_info.form
-            assert form.validate() is False
-            
+                StepPauschbetragPersonA.name, True, ImmutableMultiDict(data))
+            assert step.redirection_step_name == StepDisabilityPersonA.name
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:29]: We shouldn't test this here but in the functional test

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py [hunk 26]
```diff
@@ -16,31 +20,28 @@ def test_if_person_a_requests_pauschbetrag_is_given_then_validation_should_be_su
                 StepPauschbetragPersonA.name, True, ImmutableMultiDict(form_data))
             form = step.render_info.form
             assert form.validate() is True
-            
-    def test_if_person_a_requests_pauschbetrag_is_not_given_then_validation_should_be_false(self, new_test_request_context):
-        form_data = {}
-        stored_data = {'person_a_has_disability': 'yes', 'person_a_has_pflegegrad': 'yes'}
-        with new_test_request_context(form_data=form_data, stored_data=stored_data, method='POST'):
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:23]: Why did you delete this test?

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py [hunk 27]
```diff
@@ -16,31 +20,28 @@ def test_if_person_a_requests_pauschbetrag_is_given_then_validation_should_be_su
...
+            assert step.redirection_step_name == StepDisabilityPersonA.name
 
 class TestPreconditionPauschbetragPersonAValidation:
     def test_if_empty_data_then_redirect_to_person_a_has_disability(self, new_test_request_context):
         data = MultiDict({})
         with new_test_request_context(stored_data=data):
             step = LotseStepChooser().get_correct_step(
                 StepPauschbetragPersonA.name, True, ImmutableMultiDict(data))
-            assert step.redirection_step_name == 'has_disability_person_a'
+            assert step.redirection_step_name == StepDisabilityPersonA.name
```
Comment (JulianRoesner) [inline webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:37]: 👍

File: webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py (hunk 25 above)
Comment (JulianRoesner) [reply webapp/tests/forms/steps/lotse/test_pauschbetrag_steps.py:29]: There are a lot more tests like this in this file. Maybe we could have another look at those, too?

File: webapp/tests/model/test_form_data.py [hunk 28]
```diff
@@ -276,28 +300,73 @@ def test_if_no_steuernummer_and_no_new_tax_number_request_then_raise_missing_err
...
+                   MagicMock(side_effect=ValidationError([], HasPauschbetragClaimPersonBPrecondition))):
+            MandatoryFormData.parse_obj({**self.valid_data_person_a, **self.valid_data_person_b,
+                                         **self.valid_steuernummer, **self.valid_married_familienstand})
+
+    def test_if_person_b_requests_pauschbetrag_claim_then_raise_error_if_requests_pauschbetrag_is_missing(self):
+        with patch('app.forms.steps.lotse.pauschbetrag.HasPauschbetragClaimPersonBPrecondition.__init__',
+                   MagicMock(return_value=None)):
+            with self.assertRaises(ValidationError) as validation_error:
+                MandatoryFormData.parse_obj({**self.valid_data_person_a, **self.valid_data_person_b,
+                                             **self.valid_steuernummer, **self.valid_married_familienstand})
```
Comment (JulianRoesner) [inline webapp/tests/model/test_form_data.py:364]: We need to add similar behavior to #536

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 29]
```diff
@@ -209,9 +342,64 @@ def render(self):
...
+        ).camelized_dict()
+
+        return render_template('react_component.html',
+                               component='NoPauschbetragPage',
+                               props=props_dict,
+                               form=self.render_info.form,
+                               header_title=_('form.lotse.header-title'))
+
+
+class StepNoPauschbetragPersonB(LotseFormSteuerlotseStep):
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:386]: I am wondering if we should put those classes into their own files. WDYT?

**Review by JulianRoesner — CHANGES_REQUESTED** [review]
Still some minor changes. Mostly because in the tests

File: webapp/app/translations/de/LC_MESSAGES/messages.po [hunk 30]
```diff
@@ -1292,6 +1293,33 @@ msgstr "Angaben zu Behinderung oder Pflegebedürftigkeit von Person B"
...
 msgstr "Angaben zur festgestellen Behinderung oder Pflegebedürftigkeit Person B"
 
+#: app/forms/steps/lotse/pauschbetrag.py:92
+#: app/forms/steps/lotse/pauschbetrag.py:99
+msgid "form.lotse.skip_reason.has_not_filled_pflegegrad"
+msgstr ""
+"Sie haben angegeben, dass Sie keine Angaben zu einer Behinderung oder "
+"Pflegebedürftigkeit machen möchten. Ändern Sie die Antwort der Frage, "
+"wenn sie doch Angaben zu einer Behinderung oder Pflegebedürftigkeit "
+"machen möchten."
```
[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/app/translations/de/LC_MESSAGES/messages.po]: Done

File: webapp/app/forms/flows/lotse_flow.py [hunk 31]
```diff
@@ -368,12 +365,20 @@ def _collect_data_for_step(self, step, form_data, missing_fields=None):
...
                 field = getattr(step.form, attr)
                 label = field.kwargs['render_kw']['data_label']
                 step_data[label] = _l('form.lotse.missing_mandatory_field')
+            elif hasattr(step.form, attr):
+                field = getattr(step.form, attr)
+                # TODO: When the summary page is refactored we should merge _generate_value_representation & get_overview_value_representation
+                label, value = self._generate_value_representation(field, form_data.get(attr))
+                value = step.get_overview_value_representation(value)                
+                if value is not None:
+                    step_data[label] = value
```
Comment (JulianRoesner) [inline webapp/app/forms/flows/lotse_flow.py:381]: What is the difference between this and lines 365 to 370?

File: webapp/app/forms/steps/lotse/merkzeichen.py [hunk 32]
```diff
@@ -124,21 +120,16 @@ class InputForm(SteuerlotseBaseForm):
...
-            render_kw={'data_label': _l('form.lotse.merkzeichen.has_merkzeichen_bl.data_label')},
-            name='person_b_has_merkzeichen_bl')
-        person_b_has_merkzeichen_tbl = BooleanField(
-            render_kw={'data_label': _l('form.lotse.merkzeichen.has_merkzeichen_tbl.data_label')},
-            name='person_b_has_merkzeichen_tbl')
-        person_b_has_merkzeichen_h = BooleanField(
-            render_kw={'data_label': _l('form.lotse.merkzeichen.has_merkzeichen_h.data_label')},
-            name='person_b_has_merkzeichen_h')
+        person_b_has_merkzeichen_g = MerkzeichenBooleanField(
+            render_kw={'data_label': _l('form.loMerkzeichenBooleanFieldtse.merkzeichen.has_merkzeichen_g.data_label')})
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/merkzeichen.py:124]: I think that was a typo:
```suggestion
            render_kw={'data_label': _l('form.lotse.merkzeichen.has_merkzeichen_g.data_label')})
```

File: webapp/app/forms/steps/lotse/pauschbetrag.py [hunk 33]
```diff
@@ -52,19 +56,130 @@ def calculate_pauschbetrag(has_pflegegrad=False, disability_degree=None, has_mer
...
+            else None
+        values['person_b_disability_degree'] = int(values['person_b_disability_degree']) \
+            if values['person_b_disability_degree'] \
+            else None
+        return values
+
+
+class HasPauschbetragClaimPersonAPrecondition(DisabilityModel):
+    _step_to_redirect_to = StepMerkzeichenPersonA.name
+    _message_to_flash = _l('form.lotse.skip_reason.has_no_pauschbetrag_claim')
```
Comment (JulianRoesner) [inline webapp/app/forms/steps/lotse/pauschbetrag.py:95]: Shouldn't this be a different message?
```suggestion
    _message_to_flash = _l('form.lotse.skip_reason.has_pauschbetrag_claim')
```

We probably need to add that one to messages.po, too

**Review by SannyNguyenHung — APPROVED** [review]

File: webapp/client/src/components/FormFieldRadioGroup.js [hunk 34]
```diff
@@ -19,7 +19,6 @@ const Radio = styled.div`
   }
 
   input[type="radio"] + label {
-    white-space: pre-wrap;
```
Comment (JulianRoesner) [inline webapp/client/src/components/FormFieldRadioGroup.js:22]: You added this line with #516. I guess that it was copied to here. I am not so sure why it was added (that was the reason for my question) but I am wondering if we can just delete it. I guess it was added to fulfill some design criteria as this [deals with new lines](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space)

**Review by JulianRoesner — APPROVED** [review]
I added some small changes. Besides that, I have only some open questions. Maybe let's talk about it on Monday?

File: webapp/tests/forms/steps/lotse/test_confirmation.py [hunk 35]
```diff
@@ -48,8 +48,7 @@ def test_set_section_steps_in_render_info(self, new_test_request_context, summar
     def test_if_data_missing_then_set_next_url_correct(self, new_test_request_context, summary_step):
         data_with_missing_fields = {'steuernummer': 'C3P0'}
 
-        with new_test_request_context(stored_data=data_with_missing_fields), \
-                patch('app.forms.flows.lotse_flow.LotseMultiStepFlow._get_overview_data'):
```
[the remark this answers is not in the document]
Comment (JulianRoesner) [reply webapp/tests/forms/steps/lotse/test_confirmation.py:52]: Because we had changes in there summary page and now need the result

File: webapp/client/src/components/FormFieldRadioGroup.js (hunk 34 above)
Comment (JulianRoesner) [reply webapp/client/src/components/FormFieldRadioGroup.js:22]: This was added because there was a strange behavior in the radio field when using a Trans object. This was fixed in a different branch and can be deleted.
